hashintel/hash
# Migrate JS internal packages to TS

## 🌟 What is the purpose of this PR?

The PR aims to provide support for Typescript.
This means having Typescript packages and behaviors.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1199548034582004/1202239593614677/f) _(internal)_

## 🚫 Blocked by

- [ ] [Investigate memory leak](https://app.asana.com/0/1199548034582004/1202479511750422/f) (_internal_)

## 🔍 What does this change?

- Allow packages with the extensions "js", "jsx", "ts" and "tsx"

## 📜 Does this require a change to the docs?



- The docs for x need updating to explain that y

## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?

- All tests involving Javascript behaviors

## ❓ How to test this?

1.  Checkout the branch / view the deployment
2.  Try `cargo test`
3.  Confirm that all tests pass


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203358502199055
